core/types/types.cc

```diff
@@ -333,7 +347,8 @@ TypePtr Types::tClass(const TypePtr &attachedClass) {
 }
 
 TypePtr Types::dropNil(const GlobalState &gs, const TypePtr &from) {
-    return Types::dropSubtypesOf(gs, from, Symbols::NilClass());
+    static InlinedVector<ClassOrModuleRef, 1> toDrop{core::Symbols::NilClass()};
+    return Types::dropSubtypesOf(gs, from, toDrop);
```

Comment:
Optionally could just `auto toDrop = absl::make_span(&core::Symbols::NilClass(), 1);` or something like that, which ought to be epsilon faster.